src/pepr/operator/controllers/keycloak/authservice/authservice.spec.ts

```diff
@@ -26,45 +26,57 @@ vi.mock("../../istio/ambient-waypoint", () => ({
   setupAmbientWaypoint: vi.fn().mockResolvedValue(undefined),
 }));
 
-// Mock the K8s API
+// --- Pepr K8s and R utility mock ---
 const mockGet = vi.fn();
 const mockUpdate = vi.fn();
 const mockApply = vi.fn();
 const mockList = vi.fn();
 
-vi.mock("kubernetes-fluent-client", async importOriginal => {
-  const actual = await importOriginal<typeof import("kubernetes-fluent-client")>();
-  return {
-    ...actual,
-    K8s: {
-      withNamespace: () => ({
-        Get: mockGet,
-        Update: mockUpdate,
-        Apply: mockApply,
-        List: mockList,
-      }),
-      IstioAuthorizationPolicy: {
-        withNamespace: () => ({
-          Apply: vi.fn().mockResolvedValue({}),
-          List: vi.fn().mockResolvedValue({ items: [] }),
-        }),
-      },
-      IstioRequestAuthentication: {
-        withNamespace: () => ({
-          Apply: vi.fn().mockResolvedValue({}),
-          List: vi.fn().mockResolvedValue({ items: [] }),
-        }),
-      },
-      Namespace: {
-        Get: vi.fn().mockResolvedValue({
-          metadata: {
-            name: "test-ns",
-          },
+vi.mock("pepr", () => ({
+  K8s: () =>
+    Object.assign({
+      Get: mockGet,
+      Update: mockUpdate,
+      Apply: mockApply,
+      List: mockList,
+      // Add more methods if needed
+      InNamespace: vi.fn().mockImplementation(() => ({
+        WithLabel: vi.fn().mockReturnThis(),
+        Delete: vi.fn().mockResolvedValue(undefined),
+        Get: vi.fn().mockResolvedValue({ items: [] }),
+        // Add other chainable/query methods as needed
+      })),
+    }),
+  kind: {
+    Secret: "Secret",
+    // Add more kinds if needed
+  },
+  R: {
+    filter: <T>(fn: (value: T, index: number, array: T[]) => boolean, arr: T[]): T[] =>
```

Comment:
I don't think we should need to mock `R` here, might just need to use the `actual` like in this example: https://github.com/defenseunicorns/uds-core/blob/8686ceb69306e049c20cd50fe422132f7c42a1ff/src/pepr/prometheus/utils.spec.ts#L12-L22